distributed/shuffle/_arrow.py

```diff
@@ -28,6 +28,27 @@ def check_dtype_support(meta_input: pd.DataFrame) -> None:
             raise TypeError("p2p does not support sparse data found in column '{name}'")
 
 
+def check_minimal_arrow_version() -> None:
+    """Verify that the the correct version of pyarrow is installed to support
+    the P2P extension.
+
+    Raises a RuntimeError in case pyarrow is not installed or installed version
+    is not recent enough.
+    """
+    suffix = ""
+    # First version to introduce Table.sort_by
+    minversion = "7.0.0"
+    try:
+        import pyarrow as pa
+        from packaging.version import parse
+
+        if parse(pa.__version__) >= parse(minversion):
+            return
+        suffix = f" but only found {pa.__version__}"
+    finally:
```

Comment:
This is triggered even on the early return above. This logic needs slight fixing to only trigger on bad versions.

Reply:
oh, right :D